src/dump.ts

```diff
@@ -59,4 +63,64 @@ namespace Il2Cpp {
         file.close();
         ok(`dump saved to ${destination}`);
     }
+
+    export function dumpTree(path?: string, deleteIfExists: boolean = false): void {
+        const basePath = path ?? `${Il2Cpp.application.dataPath!}/${Il2Cpp.application.identifier ?? "unknown"}_${Il2Cpp.application.version ?? "unknown"}`;
+        const basePathExists = directoryExists(basePath);
+
+        if (!deleteIfExists && basePathExists) {
+            warn(`directory ${basePath} already exists, skipping...`);
+            return;
+        }
+
+        if (deleteIfExists && basePathExists) {
+            warn(`directory ${basePath} already exists, but tree deletion not yet supported, skipping...`);
+            return;
+        }
+
+        for (const assembly of Il2Cpp.domain.assemblies) {
+            const assemblyParts = assembly.name.split(".");
+            const assemblyPath = assemblyParts.length >= 2 ? assemblyParts.slice(0, -1).join("/") : null;
+            const filename = assemblyParts[assemblyParts.length - 1] + '.cs';
+            const path = assemblyPath ? `${basePath}/${assemblyPath}` : basePath;
+
+            // Create directory (recursively) if necessary
+            createDirectoryRecursively(path);
+
+            const destination = `${path}/${filename}`;    
```

Comment:
```suggestion
        for (const assembly of Il2Cpp.domain.assemblies) {
            const destination = assembly.name.replaceAll(".", "/") + ".cs";

            createDirectoryRecursively(destination.substring(0, destination.lastIndexOf("/")));
```

Reply:
Almost, still needs to include the `basePath`, but I'll simplify a bit along the lines of your flavour.